e2e/util/kustomize.go

```diff
@@ -0,0 +1,225 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// buildKustomization builds the kustomization and returns YAML bytes
+func buildKustomization(kustomizationPath string) ([]byte, error) {
+	fSys := filesys.MakeFsOnDisk()
+
+	options := krusty.MakeDefaultOptions()
+	k := krusty.MakeKustomizer(options)
+
+	resMap, err := k.Run(fSys, kustomizationPath)
+	if err != nil {
+		return nil, fmt.Errorf("kustomize build failed: %w", err)
+	}
+
+	yaml, err := resMap.AsYaml()
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to YAML: %w", err)
+	}
+
+	return yaml, nil
+}
+
+// createK8sClient creates a dynamic Kubernetes client and REST mapper
+func createK8sClient(kubeconfig string) (dynamic.Interface, meta.RESTMapper, error) {
+	config, err := clientcmd.LoadFromFile(kubeconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to load kubeconfig: %w", err)
+	}
+
+	restConfig, err := clientcmd.NewDefaultClientConfig(*config, nil).ClientConfig()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create REST config: %w", err)
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create dynamic client: %w", err)
+	}
+
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(restConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create discovery client: %w", err)
+	}
+
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
+
+	return dynamicClient, mapper, nil
+}
+
+// ApplyKustomization builds and applies a kustomization using Kubernetes API
+func ApplyKustomization(kustomizationPath, namespace, kubeconfig string) error {
```

Comment:
Public functions(ApplyKustomization and DeleteKustomization) should be at the top, followed by privates.